src/wallet/wallet.cpp

```diff
@@ -1806,6 +1820,8 @@ CWallet::ScanResult CWallet::ScanForWalletTransactions(const uint256& start_bloc
                 WalletLogPrintf("Still rescanning. At block %d. Progress=%f\n", *block_height, progress_current);
             }
 
+            if (chain().filterMatchesAny(block_hash, filter_set)) {
```

Comment:
`result.last_scanned_block` and `result.last_scanned_height` should both be updated even if the filter doesn't match.

Reply:
Oh, there's another optimization possible, which is to not call `filterMatchesAny` if the filter index doesn't exist or doesn't contain any entry for any of the previously queried block hashes. If `filterMatchesAny` indicates whether the failure is due to that or due to the filter existing but not matching, it would be pretty easy to check this.

Reply:
Indeed, last_scanned_height is reported by rpc as the stop_height, so should be updated.